internal/batches/executor/coordinator.go

```diff
@@ -177,7 +182,7 @@ func (c *Coordinator) buildSpecs(ctx context.Context, batchSpec *batcheslib.Batc
 
 // ExecuteAndBuildSpecs executes the given tasks and builds changeset specs for the results.
 // It calls the ui on updates.
-func (c *Coordinator) ExecuteAndBuildSpecs(ctx context.Context, batchSpec *batcheslib.BatchSpec, tasks []*Task, ui TaskExecutionUI) ([]*batcheslib.ChangesetSpec, []string, error) {
+func (c *Coordinator) ExecuteAndBuildSpecs(ctx CancelableContext, batchSpec *batcheslib.BatchSpec, tasks []*Task, ui TaskExecutionUI) ([]*batcheslib.ChangesetSpec, []string, error) {
```

Comment:
Since the arguments to this method is getting bulky, we can move it into a struct and pass in the cancel function explicitly.

```suggestion
type ExecuteAndBuildSpecsArgs struct {
	batchSpec *batcheslib.BatchSpec
	tasks []*Task
	ui TaskExecutionUI
	cancelFunc context.CancelCauseFunc
}

func (c *Coordinator) ExecuteAndBuildSpecs(ctx context.Context, args ExecuteAndBuildSpecsArgs) ([]*batcheslib.ChangesetSpec, []string, error) {
```

Reply:
The cancelFunc isn't being used here, I wonder if we need to pass it in here as an argument?